src/wp-includes/class-wp-scripts.php

```diff
@@ -295,14 +304,34 @@ public function do_item( $handle, $group = false ) {
 		}
 
 		$before_handle = $this->print_inline_script( $handle, 'before', false );
-		$after_handle  = $this->print_inline_script( $handle, 'after', false );
 
 		if ( $before_handle ) {
 			$before_handle = sprintf( "<script%s id='%s-js-before'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $before_handle );
 		}
 
+		// Eligible loading strategies will only be 'async', 'defer', or ''.
```

Comment:
Let's remove this comment. We should also adjust the code to consider specific loading strategies rather than just `'' !== $strategy`. Considering any non-empty strategy for the extra logic here is error-prone considering that more strategies may be added in the future, which was an important consideration to make this enhancement about a "script loading strategy" rather than just "support async/defer".

By explicitly running the relevant logic only when `$strategy` is either "defer" or "async", we ensure we don't accidentally run it for other strategies - yes, those don't exist now, but they could be added later. When another WordPress core PR may be opened in the future with another strategy, the author of that PR will have it way easier if we run logic for the specific strategies:
* If they add a new strategy, by default it won't behave differently from the default strategy. So we make no assumptions on other strategies, which is good.
* If they want to reuse some of the logic that applies to e.g. the `defer` or `async` strategy, they can easily search for where in this class those checks happen and then for example add their own strategies to that same list to check.